lib/grape/dsl/request_response.rb

```diff
@@ -105,7 +105,13 @@ def rescue_from(*args, &block)
           end
 
           options = args.extract_options!
-          handler ||= proc { options[:with] } if options.key?(:with)
+          if options.key?(:with)
+            if options[:with].instance_of?(Proc)
+              handler ||= options[:with]
+            elsif !(options[:with].instance_of?(Symbol) || options[:with].instance_of?(String))
+              fail ArgumentError, ":with option expected Symbol, String or Proc, got #{options[:with].class}"
```

Comment:
I would turn this around, "with: #{options[:with].class}, expected Symbol, String or Proc".